File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.py
def bracket(s):
    stack = []
    for i in s:
        if len(stack) == 0: stack.append(i)
        elif stack[-1] == "(" and i == ")": stack.pop()
        elif stack[-1] == "[" and i == "]": stack.pop()
        elif stack[-1] == "{" and i == "}": stack.pop()
        else:
            stack.append(i)
    if len(stack) == 0:
        return 1
    return 0



def solution(s):
    answer = 0

    for i in range(len(s)):
        if bracket(s):  answer += 1
        s = s[1:] + s[:1]
    print(answer)
    return answer

File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;
class Solution {
    public int solution(String s) {
        int answer = 0;
        for (int i = 0; i < s.length(); i++) {
            if (check(s)) {
                answer++;
            }
            String temp = String.valueOf(s.charAt(0));
            s = s.substring(1, s.length()) + temp;
        }
        return answer;
    }

    public boolean check(String s) {
        Stack<Character> stack = new Stack<>();

        for (int i = 0; i < s.length(); i++) {
            if (stack.size() == 0) {
                stack.add(s.charAt(i));
            } else if (s.charAt(i) == ']' && stack.peek() == '[') {
                stack.pop();
            } else if (s.charAt(i) == '}' && stack.peek() == '{') {
                stack.pop();
            } else if (s.charAt(i) == ')' && stack.peek() == '(') {
                stack.pop();
            } else {
                stack.add(s.charAt(i));
            }
        }
        return stack.isEmpty();
    }
}